File: apps/web/src/types/importCase.ts
export interface ImportCaseSummary {
  id: string;
  caseNumber: string;
  title: string;
  status: string;
  derivedOperationalStatus: string;
  supplierName: string;
  defaultTransportMode?: string;
  incoterm?: string;
  productionStatus: string;
  responsibleUserName?: string;
  estimatedProductionCompletionDate?: string;
  readyForShipmentDate?: string;
  minEtd?: string;
  maxEta?: string;
  isDelayed: boolean;
  lineCount: number;
  shipmentCount: number;
  createdAtUtc: string;
  updatedAtUtc: string;
}

export interface ImportCaseDetail {
  id: string;
  caseNumber: string;
  title: string;
  status: string;
  derivedOperationalStatus: string;
  supplierName: string;
  originCountry?: string;
  defaultTransportMode?: string;
  incoterm?: string;
  responsibleUserId?: string;
  responsibleUserName?: string;
  purchasingOwnerUserId?: string;
  purchasingOwnerUserName?: string;
  operationsOwnerUserId?: string;
  operationsOwnerUserName?: string;
  productionStatus: string;
  estimatedProductionCompletionDate?: string;
  readyForShipmentDate?: string;
  notes?: string;
  closedAtUtc?: string;
  createdAtUtc: string;
  updatedAtUtc: string;
  rowVersion: number;
  lines: ImportCaseLine[];
  shipments: ShipmentSummary[];
}

export interface ImportCaseLine {
  id: string;
  importCaseId: string;
  purchaseOrderLineId: string;
  orderNumber: string;
  lineNumber: number;
  stockCode: string;
  stockName: string;
  orderedQuantity: number;
  allocatedQuantity: number;
  releasedQuantity: number;
  effectiveAllocatedQuantity: number;
  shippedQuantity: number;
  receivedQuantity: number;
  status: string;
  plannedShipmentDate?: string;
  notes?: string;
  createdAtUtc: string;
  updatedAtUtc: string;
  rowVersion: number;
}

export interface AvailablePurchaseOrderLine {
  purchaseOrderLineId: string;
  purchaseOrderId: string;
  orderNumber: string;
  lineNumber: number;
  stockCode: string;
  stockName: string;
  supplierName: string;
  orderDate: string;
  orderedQuantity: number;
  remainingQuantity: number;
  allocatedToOtherCases: number;
  effectiveAvailableQuantity: number;
}

export interface ShipmentSummary {
  id: string;
  importCaseId: string;
  shipmentSequence: number;
  shipmentNumber: string;
  transportMode: string;
  originLocation: string;
  destinationLocation: string;
  originTimezoneId: string;
  destinationTimezoneId: string;
  bookingNumber?: string;
  forwarderName?: string;
  carrierName?: string;
  transportReference?: string;
  vesselName?: string;
  voyageNumber?: string;
  etd?: string;
  eta?: string;
  atd?: string;
  ata?: string;
  status: string;
  containerCount: number;
  lineAllocationCount: number;
  createdAtUtc: string;
  updatedAtUtc: string;
}

export interface ShipmentDetail extends ShipmentSummary {
  caseNumber: string;
  estimatedWarehouseArrival?: string;
  actualWarehouseArrival?: string;
  notes?: string;
  rowVersion: number;
  lineAllocations: ShipmentLineAllocation[];
  containers: ShipmentContainer[];
  milestones: ShipmentMilestone[];
}

export interface ShipmentLineAllocation {
  id: string;
  shipmentId: string;
  importCaseLineId: string;
  importCaseId: string;
  stockCode: string;
  stockName: string;
  caseAllocatedQuantity: number;
  allocatedQuantity: number;
  releasedQuantity: number;
  effectiveAllocatedQuantity: number;
  shippedQuantity: number;
  receivedQuantity: number;
  status: string;
  createdAtUtc: string;
  updatedAtUtc: string;
  rowVersion: number;
}

export interface ShipmentContainer {
  id: string;
  shipmentId: string;
  containerNumber: string;
  normalizedContainerNumber: string;
  containerType: string;
  sealNumber?: string;
  grossWeightKg?: number;
  netWeightKg?: number;
  packageCount?: number;
  status: string;
  notes?: string;
  createdAtUtc: string;
  updatedAtUtc: string;
  rowVersion: number;
}

export interface ShipmentMilestone {
  id: string;
  shipmentId: string;
  sequenceNumber: number;
  milestoneType: string;
  locationName?: string;
  timezoneId: string;
  plannedAtUtc?: string;
  estimatedAtUtc?: string;
  actualAtUtc?: string;
  status: string;
  source: string;
  notes?: string;
  createdAtUtc: string;
  updatedAtUtc: string;
  rowVersion: number;
}

export interface SupplierLookup {
  supplierName: string;
  normalizedSupplierName: string;
  activeOrderCount: number;
}

export interface ImportCaseOperationalSummary {
  activeCaseCount: number;
  productionDelayedCount: number;
  readyForShipmentCount: number;
  bookingPendingCount: number;
  inTransitShipmentCount: number;
  delayedShipmentCount: number;
  etaThisWeekCount: number;
  unallocatedLineCount: number;
}

File: apps/api/src/ImportControlTower.Domain/Entities/ImportCase.cs
using System;
using System.Collections.Generic;

namespace ImportControlTower.Domain.Entities;

public class ImportCase
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string CaseNumber { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Status { get; set; } = "Draft";
    public string SupplierName { get; set; } = string.Empty;
    public string NormalizedSupplierName { get; set; } = string.Empty;
    public string? OriginCountry { get; set; }
    public string? DefaultTransportMode { get; set; }
    public string? Incoterm { get; set; }
    public Guid? ResponsibleUserId { get; set; }
    public ApplicationUser? ResponsibleUser { get; set; }
    public Guid? PurchasingOwnerUserId { get; set; }
    public ApplicationUser? PurchasingOwnerUser { get; set; }
    public Guid? OperationsOwnerUserId { get; set; }
    public ApplicationUser? OperationsOwnerUser { get; set; }
    public string? Notes { get; set; }
    public string ProductionStatus { get; set; } = "NotStarted";
    public DateTime? EstimatedProductionCompletionDate { get; set; }
    public DateTime? ReadyForShipmentDate { get; set; }
    public int LastShipmentSequence { get; set; } = 0;
    public DateTime? ClosedAtUtc { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;
    public Guid CreatedByUserId { get; set; }
    public ApplicationUser? CreatedByUser { get; set; }
    public Guid? UpdatedByUserId { get; set; }
    public ApplicationUser? UpdatedByUser { get; set; }

    public ICollection<ImportCaseLine> Lines { get; set; } = new List<ImportCaseLine>();
    public ICollection<Shipment> Shipments { get; set; } = new List<Shipment>();
}
